admin/src/bench/mod.rs

```diff
@@ -30,40 +32,97 @@ static COMMITS: AtomicUsize = AtomicUsize::new(0);
 //static QUERIES: AtomicUsize = AtomicUsize::new(0);
 
 const COMMIT_SIZE: usize = 100;
+// Note on db, this is equal to COMMIT_SIZE.
+// Having smaller allow validating a given amount
+// of keys.
 const COMMIT_PRUNE_SIZE: usize = 90;
 const COMMIT_PRUNE_WINDOW: usize = 2000;
 
-const USAGE: &str = "
-Usage: stress [--readers=<#>][--writers=<#>] [--commits=<l>]
+pub(super) struct BenchAdapter(parity_db::Db);
+
+impl BenchDb for BenchAdapter {
+	type Options = parity_db::Options;
+
+	fn open(path: &std::path::Path) -> Self {
+		BenchAdapter(Db::with_columns(path, 1).unwrap())
+	}
+
+	fn with_options(options: &Self::Options) -> Self {
+		BenchAdapter(Db::open_or_create(options).unwrap())
+	}
+
+	fn get(&self, key: &Key) -> Option<Value> {
+		self.0.get(0, key).unwrap()
+	}
+
+	fn commit<I: IntoIterator<Item=(Key, Option<Value>)>>(&self, tx: I) {
+		self.0.commit(tx.into_iter().map(|(k, v)| (0, k, v))).unwrap()
+	}
+}
+
+/// Stress tests (warning erase db first).
+#[derive(Debug, StructOpt)]
+pub struct Stress {
+	#[structopt(flatten)]
+	pub shared: Shared,
+
+	/// Number of reading threads [default: 4].
+	#[structopt(long)]
+	pub readers: Option<usize>,
+
+	/// Number of writing threads [default: 1].
+	#[structopt(long)]
+	pub writers: Option<usize>,
+
+	/// Start commit index.
+	#[structopt(long)]
+	pub start_commit: Option<usize>,
+
 
-Options:
-	--readers=<#>      Number of reading threads [default: 4].
-	--writers=<#>      Number of writing threads [default: 1].
-	--commits=<n>      Total numbet of inserted commits.
-	--seed=<n>         Random seed used for key generation.
-	--archive          Do not perform deletions.
-	--append           Open an existing database.
-";
+	/// Total number of inserted commits.
+	#[structopt(long)]
+	pub commits: Option<usize>,
+
+	/// Random seed used for key generation.
+	#[structopt(long)]
+	pub seed: Option<u64>,
+
+	/// Open an existing database.
+	#[structopt(long)]
+	pub append: bool,
+
+	/// Do not apply pruning.
+	#[structopt(long)]
+	pub archive: bool,
+
+	/// Do only check (requires append flag).
+	#[structopt(long)]
+	pub check_only: bool,
+}
 
 #[derive(Clone)]
-struct Args {
-	readers: usize,
-	commits: usize,
-	writers: usize,
-	seed: Option<u64>,
-	archive: bool,
-	append: bool,
+pub struct Args { // TODO remove (rendundant with Stress)
+	pub readers: usize,
+	pub commits: usize,
+	pub writers: usize,
+	pub seed: Option<u64>,
+	pub archive: bool,
+	pub append: bool,
+	pub check_only: bool,
+	pub start_commit: usize,
 }
 
-impl Default for Args {
-	fn default() -> Args {
+impl Stress {
+	pub(super) fn get_args(&self) -> Args {
 		Args {
-			readers: 4,
-			writers: 1,
-			commits: 100000,
-			append: false,
-			seed: None,
-			archive: false,
+			readers: self.readers.unwrap_or(4),
+			writers: self.writers.unwrap_or(1),
+			commits: self.commits.unwrap_or(100_000),
+			seed: self.seed.clone(),
+			append: self.append,
+			archive: self.archive,
+			check_only: self.check_only,
```

Comment:
actually check_only and start_commit looks new

Reply:
I was thinking we should actually put the last commit number in the database along with key values. Also put the seed in the first commit. This would allow for correct continuation with `--append`